apps/src/templates/sectionProgress/progressTables/ProgressTableLevelBubble.jsx

```diff
@@ -62,28 +73,81 @@ class ProgressTableLevelBubble extends React.PureComponent {
     ) : null;
   }
 
-  render() {
-    const {isUnplugged, isConcept, isDisabled, bubbleSize} = this.props;
+  isEnabled() {
+    return (
+      !this.props.isDisabled && this.props.levelStatus !== LevelStatus.locked
+    );
+  }
+
+  /**
+   * The only variation in the visual appearance of a bubble for a given level
+   * comes from `levelStatus`, so we cache bubbles by levelId and status.
+   */
+  cacheKey() {
+    return `levelId=${this.props.levelId}&status=${this.props.levelStatus}`;
```

Comment:
Did you consider taking it one step further and caching by bubble shape and level number instead of `levelId`? I think this is a good starting point, I'm just curious.

Reply:
🎉